torch/distributed/distributed_c10d.py

```diff
@@ -1410,7 +1435,7 @@ def all_gather_object(object_list, obj, group=group.WORLD):
     # Resize tensor to max size across all ranks.
     input_tensor.resize_(max_object_size)
     coalesced_output_tensor = torch.empty(
-        max_object_size * group_size, dtype=torch.uint8, device=current_device
+        max_object_size * group_size, dtype=torch.uint8, device=current_device  # type: ignore[arg-type]
```

Comment:
If is_nccl_backend is set, current_device is an int, which will not be accepted by torch.empty().